atomata/capstone_WebClient
# Enhance the editor to accept an already existing experience file or create a new experience

### Summary 
This PR enhances the editor page to start up for a new experience or to load an already existing experience for modification. This PR also enhances the UI look and fixes some of the UI bugs.
### Details
- added abstraction to how the overlay recieves a data ( experience or apparatus json for a new experience) 
- fixes some of the UI bugs that caused UI buttons color change after clicking certain buttons 
### How to test?
- run the project. select "create a new experience) -> select an apparatus -> create a sequence of actions ( delete , add, reorder)  -> click on save button in navbar -> click on return button in navbar 
- select "load an existing experience) -> in the loaded editor page you should see the proper apparatus loading and the action sequencce box pre populated with the actions you saved in the previous step
### Checks
